tests/unit/amo/pages/TestUserProfile.js

```diff
@@ -102,488 +94,438 @@ describe(__filename, () => {
     );
   }
 
-  function getHeaderPropComponent(root) {
-    const headerContent = root.find('.UserProfile-user-info').prop('header');
-    return shallow(<div>{headerContent}</div>);
-  }
+  const createErrorHandlerId = ({ userId = null }) => {
+    return `src/amo/pages/UserProfile/index.js-${extractId({
+      match: { params: { userId } },
+    })}`;
+  };
 
   it('renders user profile page', () => {
-    const root = renderUserProfile();
+    renderUserProfile();
 
-    expect(root.find('.UserProfile')).toHaveLength(1);
+    expect(screen.getByClassName('UserProfile')).toBeInTheDocument();
   });
 
   it('dispatches fetchUserAccount action if userId is not found', () => {
-    const { store } = signInUserWithUserId(100);
-    const dispatchSpy = sinon.spy(store, 'dispatch');
-    const userId = 200;
+    signInUserWithProps({ userId: 100 });
+    const dispatch = jest.spyOn(store, 'dispatch');
+    const userId = '200';
 
-    const root = renderUserProfile({ params: { userId }, store });
+    renderUserProfile({ params: { userId } });
 
-    sinon.assert.calledWith(
-      dispatchSpy,
+    expect(dispatch).toHaveBeenCalledWith(
       fetchUserAccount({
-        errorHandlerId: root.instance().props.errorHandler.id,
+        errorHandlerId: createErrorHandlerId({ userId }),
         userId,
       }),
     );
   });
 
-  it('dispatches fetchUserAccount action if userId param changes', () => {
-    const { params, store } = signInUserWithUserId(100);
-    const dispatchSpy = sinon.spy(store, 'dispatch');
+  // it('dispatches fetchUserAccount action if userId param changes', () => {
+  //   const { params } = signInUserWithProps({ userId: 100 });
+  //   const dispatch = jest.spyOn(store, 'dispatch');
 
-    const root = renderUserProfile({ params, store });
+  //   renderUserProfile({ params });
 
-    dispatchSpy.resetHistory();
+  //   dispatch.mockClear();
 
-    const userId = 200;
-    root.setProps({ match: { params: { userId } } });
+  //   const userId = 200;
+  //   root.setProps({ match: { params: { userId } } });
```

Comment:
Not really sure what to replace this `root.setProps(...)` with - and there a few other tests that have the same thing.

Reply:
Yeah, this is tricky. We cannot do `root.setProps` anymore, nor do we really want to, but we need to find something to replace it with. In some cases it is fairly simple, like just dispatching an action, but in other places it's more complicated, especially when we need to assert about a different action being dispatched. I will take a look at these.

Reply:
I haven't looked at all the tests that use `root.setProps`, so this might not be the solution for all of them, but it does fix one, and shows an example of what we need to do in this situation. We need to do *something* to either the component (i.e., interacting with it as a user would), or we need to change the state, which will trigger a re-render of the component. 

In the case of `dispatches fetchUserAccount action if userId param changes`, we can dispatch an action that would cause the component to re-render with a different userId. We would do that by changing the location, which we can do by dispatching `onLocationChanged`. Here's a diff:

```diff
diff --git a/tests/unit/amo/pages/TestUserProfile.js b/tests/unit/amo/pages/TestUserProfile.js
index 68d8da2aa..f4091991c 100644
--- a/tests/unit/amo/pages/TestUserProfile.js
+++ b/tests/unit/amo/pages/TestUserProfile.js
@@ -27,6 +27,7 @@ import {
   fakeReview,
   getElement,
   getElements,
+  onLocationChanged,
   renderPage as defaultRender,
   screen,
   within,
@@ -121,24 +122,26 @@ describe(__filename, () => {
     );
   });

-  // it('dispatches fetchUserAccount action if userId param changes', () => {
-  //   const { params } = signInUserWithProps({ userId: 100 });
-  //   const dispatch = jest.spyOn(store, 'dispatch');
-
-  //   renderUserProfile({ params });
+  it('dispatches fetchUserAccount action if userId param changes', () => {
+    const { params } = signInUserWithProps({ userId: 100 });
+    const dispatch = jest.spyOn(store, 'dispatch');

-  //   dispatch.mockClear();
+    renderUserProfile({ params });

-  //   const userId = 200;
-  //   root.setProps({ match: { params: { userId } } });
+    const userId = 200;
+    store.dispatch(
+      onLocationChanged({
+        pathname: getLocation({ userId }),
+      }),
+    );

-  //   expect(dispatch).toHaveBeenCalledWith(
-  //     fetchUserAccount({
-  //       errorHandlerId: createErrorHandlerId({ userId }),
-  //       userId: `${userId}`,
-  //     }),
-  //   );
-  // });
+    expect(dispatch).toHaveBeenCalledWith(
+      fetchUserAccount({
+        errorHandlerId: createErrorHandlerId({ userId }),
+        userId,
+      }),
+    );
+  });

   it('does not dispatch fetchUserAccount if userId does not change', () => {
     const userId = 100;
```